### DISCUSSION TITLE
AWS Sig v4 Host header

### DISCUSSION THREAD

**Discussion by frumioj** [curl/curl#10327] [General]
In [AWS Signer v4 code](https://github.com/aws/aws-sdk-go-v2/blob/b0691f2ce89932a87343c737b69d3f56f54e39f5/aws/signer/internal/v4/host.go#L9) I found this:

```
func SanitizeHostForHeader(r *http.Request) {
	host := getHost(r)
	port := portOnly(host)
	if port != "" && isDefaultPort(r.URL.Scheme, port) {
		r.Host = stripPort(host)
	}
}
```

... which results in a `Host`of `localhost:8080` for my AWS requests using the go-lang SDK. In curl, however, I found that the port is stripped from the Host header before doing AWS signature with the `--aws-sigv4` parameter. 

I can't find specification text in [Creating a Signed Request](https://docs.aws.amazon.com/general/latest/gr/create-signed-request.html) to know what the right answer is, but I am noting what appears to be divergent behaviour for this particular case, and perhaps others. 

Related discussion in the AWS SDK repo: https://github.com/aws/aws-sdk-go-v2/discussions/1992

**Comment by jay** [comment]
Ref: https://github.com/curl/curl/issues/9550

/cc @outscale-mgo @cbodley

**Reply by cbodley** [reply]
yeah this was fixed by https://github.com/curl/curl/pull/7966:
> This patch also fix a bug with host header, which was ignoring the port.
